src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSetSerializer.cs

```diff
@@ -120,14 +124,13 @@ public override void WriteObject(object graph, Type type, ODataMessageWriter mes
                     Error.Format(SRResources.TypeCannotBeSerialized, elementType.FullName()));
             }
 
-            // save this for later to support JSON odata.streaming.
-            Uri nextPageLink = resourceSet.NextPageLink;
+            // set the nextpagelink to null to support JSON odata.streaming.
             resourceSet.NextPageLink = null;
```

Comment:
>NextPageLink [](start = 24, length = 12)

So we entirely ignore any nextPageLink set explicitly on the resourceSet.
1) Does that mean we never explicitly set the NextPageLink on the resourceSet?
2) Are there scenarios where the user has access to the ResourceSet and might be explicitly setting the nextPageLink?